var/spack/repos/builtin/packages/sirius/package.py

```diff
@@ -230,6 +232,11 @@ class Sirius(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("costa+shared", when="@7.3.2:")
 
+    with when("@7.5: +memory_pool"):
+        depends_on("umpire", when=("+memory_pool"))
+        depends_on("umpire+cuda", when=("+memory_pool+cuda"))
+        depends_on("umpire+rocm", when=("+memory_pool+rocm"))
```

Comment:
```suggestion
        depends_on("umpire")
        depends_on("umpire+cuda", when="+cuda")
        depends_on("umpire+rocm", when=+rocm")
```

Then `with when(...)` constraints are added to all when conditions of directives like `depends_on` in the block, so no need to repeat them. Also `("xyz")` is a bit confusing to me :p it's the same as `"xyz"` but superficially it looks like a tuple, so maybe just use a string without brackets

Reply:
slept my mind. Thanks.